app/services/invoices/calculate_fees_service.rb

```diff
@@ -47,14 +47,15 @@ def call
         Credits::AppliedCouponsService.call(invoice:) if should_create_coupon_credit?
 
         if customer_provider_taxation?
-          taxes_result = Integrations::Aggregator::Taxes::Invoices::CreateService.call(invoice:, fees: invoice.fees)
+          taxes_result = fetch_taxes_for_invoice(invoice)
 
           unless taxes_result.success?
             create_error_detail(taxes_result.error.code)
 
-            # Draft invoice needs to stay in draft status
-            invoice.failed! if invoice.finalized?
+            # only fail invoices that are finalizing
+            invoice.failed! if finalizing_invoice?(invoice)
 
+            invoice.save
```

Comment:
This line is not needed since `invoice.failed!` saves invoice

Reply:
it will be called only for finalizing_invoices, so in case we're refreshing a draft invoice, it won't be saved, and as result new fee sums will not be applied

Reply:
okay, just call invoice.save! please